test/unix_socket/main.c

```diff
@@ -65,6 +65,67 @@ int create_connceted_sockets_default(int *sockets) {
     return create_connected_sockets(sockets, "unix_socket_default_path");
 }
 
+int create_connected_sockets_then_rename(int *sockets) {
+    char *socket_original_path = "/tmp/socket_tmp";
+    char *socket_ready_path = "/tmp/.socket_tmp";
+    int listen_fd = socket(AF_UNIX, SOCK_STREAM, 0);
+    if (listen_fd == -1) {
+        THROW_ERROR("failed to create a unix socket");
+    }
+
+    struct sockaddr_un addr;
+    memset(&addr, 0, sizeof(struct sockaddr_un)); //Clear structure
+    addr.sun_family = AF_UNIX;
+    strcpy(addr.sun_path, socket_original_path);
+    socklen_t addr_len = strlen(addr.sun_path) + sizeof(addr.sun_family);
```

Comment:
I suppose that `addr_len` should be one byte longer according to the man page:

```
       Three types of address are distinguished in the sockaddr_un
       structure:

       *  pathname: a UNIX domain socket can be bound to a null-
          terminated filesystem pathname using bind(2).  When the
          address of a pathname socket is returned (by one of the system
          calls noted above), its length is

              offsetof(struct sockaddr_un, sun_path) + strlen(sun_path)
          + 1
```

Other locations of address length calculation in the test should also be fixed.

Reply:
Got it. Done. Added a comment for notation.